File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let result = 0;
    
    function dfs(index, cal) {
        if (index === numbers.length) {
            if(cal === target) {
                result++;
            }
            return;
        }
        
        dfs(index + 1, cal + numbers[index]);
        dfs(index + 1, cal - numbers[index]);
    }
    
    dfs(0, 0);
    
    return result;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
result = 0
def dfs(index, num_sum, numbers, target):
    global result
    if index == len(numbers) and num_sum == target:
        result += 1
        return 
    if index == len(numbers):
        return
    
    dfs(index + 1, num_sum + numbers[index], numbers, target)
    dfs(index + 1, num_sum - numbers[index], numbers, target)

def solution(numbers, target):
    dfs(0, 0, numbers, target)
    return result